packages/zowe-explorer/src/trees/uss/UssFSProvider.ts

```diff
@@ -120,23 +121,39 @@ export class UssFSProvider extends BaseProvider implements vscode.FileSystemProv
 
         const oldInfo = this._getInfoFromUri(oldUri);
 
-        await ussApi.move(oldInfo.path, info.path);
+        try {
+            await ussApi.move(oldInfo.path, info.path);
+        } catch (err) {
+            this._handleError(err, {
+                additionalContext: vscode.l10n.t({ message: "Failed to move {0}", args: [oldInfo.path], comment: "File path" }),
+                apiType: ZoweExplorerApiType.Uss,
+                retry: {
+                    fn: this.move.bind(this),
+                    args: [oldUri, newUri],
+                },
+                profileType: info.profile.type,
+            });
+            throw err;
+        }
         await this._relocateEntry(oldUri, newUri, info.path);
         return true;
     }
 
     public async listFiles(profile: imperative.IProfileLoaded, uri: vscode.Uri, keepRelative: boolean = false): Promise<IZosFilesResponse> {
         const queryParams = new URLSearchParams(uri.query);
         const ussPath = queryParams.has("searchPath") ? queryParams.get("searchPath") : uri.path.substring(uri.path.indexOf("/", 1));
-        if (ussPath.length === 0) {
-            throw new imperative.ImperativeError({
-                msg: vscode.l10n.t("Could not list USS files: Empty path provided in URI"),
-            });
-        }
-        const response = await ZoweExplorerApiRegister.getUssApi(profile).fileList(ussPath);
-        // If request was successful, create directories for the path if it doesn't exist
-        if (response.success && !keepRelative && response.apiResponse.items?.[0]?.mode?.startsWith("d") && !this.exists(uri)) {
-            await vscode.workspace.fs.createDirectory(uri);
+        let response: IZosFilesResponse;
+        try {
+            response = await ZoweExplorerApiRegister.getUssApi(profile).fileList(ussPath);
+            // If request was successful, create directories for the path if it doesn't exist
+            if (response.success && !keepRelative && response.apiResponse.items?.[0]?.mode?.startsWith("d") && !this.exists(uri)) {
+                await vscode.workspace.fs.createDirectory(uri);
+            }
+        } catch (err) {
+            if (err instanceof Error) {
```

Comment:
still not a big fan of this, but I guess that's how TS want's people to handle it 😢 
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-4.html#defaulting-to-the-unknown-type-in-catch-variables---useunknownincatchvariables

Why not allow `catch(e: Error) { ... }` 💔

Reply:
big sad 😅